src/mdframe/reader.py

```diff
@@ -17,17 +16,21 @@
 MetadataFileExtension = Literal["toml"]
 SUPPORTED_METADATA_FILE_EXTENSIONS = get_args(MetadataFileExtension)
 
+# change return type here
 def load_metadata_file(metadata_file_path: Path, schema: Dict) -> Dict[str, Any]:
     if not metadata_file_path.exists():
         raise ValueError(f"Path {metadata_file_path} does not seem to point to a valid file")
 
-    with open(metadata_file_path, "r") as f:
-        metadata_file_contents = toml.load(f)
+    try:
+        with open(metadata_file_path, "r") as f:
+            metadata_file_contents = toml.load(f)
 
-    # validates JSON according to schema located in schema.json
-    validate(instance = metadata_file_contents, schema = schema)
+        # validates JSON according to schema located in schema.json
+        validate(instance = metadata_file_contents, schema = schema)
 
-    return metadata_file_contents
+        return metadata_file_contents
+    except:
+        return None        
```

Comment:
Any reason why you're returning `None` here? It would be handled better by throwing an error when the format is unsupported or even better, add a check for the file format:

```python
filename_suffix = metadata_file_path.suffix
if filename_suffix not in SUPPORTED_METADATA_FILE_EXTENSIONS:
    raise ValueError(f"{filename_suffix} is not supported")
```

If you meant catching malformed files here I would either let the `toml` library throw its error as it should be resilient enough to tell the user which part of the file could not be parsed (e.g. because the syntax was wrong)

Reply:
Sounds good, I made the error exception components more explicit (looked into the documentations for validate and toml).

I also created an invalid data file under `data_dir` in the `test` file so that we can test things out.